client/src/app/pages/applications/components/application-detail-drawer/review-fields.tsx

```diff
@@ -0,0 +1,174 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import {
+  DescriptionListGroup,
+  DescriptionListTerm,
+  DescriptionListDescription,
+  Label,
+  Title,
+  TextContent,
+} from "@patternfly/react-core";
+import { Application, Review } from "@app/api/models";
+import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+import { useFetchReviewById, useFetchReviews } from "@app/queries/reviews";
+import { useFetchArchetypes } from "@app/queries/archetypes";
+import { EmptyTextMessage } from "@app/components/EmptyTextMessage";
+import { PROPOSED_ACTION_LIST, EFFORT_ESTIMATE_LIST } from "@app/Constants";
+
+export const ReviewFields: React.FC<{ application: Application | null }> = ({
+  application,
+}) => {
+  const { archetypes } = useFetchArchetypes();
+  const { reviews } = useFetchReviews();
+  const { t } = useTranslation();
+
+  const { review: appReview } = useFetchReviewById(application?.review?.id);
+
+  const notYetReviewed = (
+    <EmptyTextMessage message={t("terms.notYetReviewed")} />
+  );
+
+  const reviewedArchetypes = application?.archetypes
+    ?.map((archetypeRef) => {
+      return archetypes.find((archetype) => archetype.id === archetypeRef.id);
+    })
+    .filter(Boolean);
+
+  const hasReviewedArchetype = reviewedArchetypes?.some(
+    (archetype) => !!archetype?.review
+  );
+
+  const matchedArchetypeReviews: Review[] = (reviewedArchetypes || [])
+    .map((reviewedArchetype) => {
+      return reviews.find(
+        (review) => review.id === reviewedArchetype?.review?.id
+      );
+    })
+    .filter(Boolean);
+
+  const groupedReviewList = [
+    {
+      review: appReview,
+      name: appReview?.application?.name,
+      type: "App",
+    },
+    ...matchedArchetypeReviews.map((archetypeReview) => ({
+      review: archetypeReview,
+      name: archetypeReview?.archetype?.name,
+      type: "Archetype",
+    })),
+  ].filter((item) => item.review?.proposedAction);
+
+  return (
+    <>
+      <TextContent className={spacing.mtLg}>
+        <Title headingLevel="h3" size="md">
+          {t("terms.review")}
+        </Title>
+      </TextContent>
+      <DescriptionListGroup>
+        <DescriptionListTerm>{t("terms.proposedAction")}</DescriptionListTerm>
+        <DescriptionListDescription cy-data="proposed-action">
+          {groupedReviewList.length > 0
+            ? groupedReviewList.map((item, index) => (
```

Comment:
Generally I like to flip the test around and put the small negative first so it is close to the test expression...

```suggestion
          {groupedReviewList.length === 0
            ? notYetReviewed
            : groupedReviewList.map((item, index) => (
```